### PR TITLE
fix: allow performing only class search in complex search

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
Backward compatible method modification that will allow change of search method so programaticaly we can decide if search should list resource OR items. 

This can be observed in item class destination placement when importing a Test

Screenshot 2024-05-22 at 10 32 31

### CODE REVIEW

**Comment by tikhanovichA** [comment]
Maybe we should use the develop branch as destination?

**Comment by tikhanovichA** [comment]
NVM, already fixed it

**Review by tikhanovichA — APPROVED** [review]
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code respects code style rules
- [x] New code respects best practices
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

File: core/kernel/persistence/smoothsql/search/ComplexSearchService.php [hunk 1]
```diff
@@ -248,9 +249,15 @@ public function getQuery(
 
         $this->setLanguage($query, $lang);
 
-        $criteria = $query->newQuery()
+        if ($onlyClass === true) {
+            $criteria = $query->newQuery()
+                ->add('http://www.w3.org/2000/01/rdf-schema#subClassOf')
```
Comment (Karol-Stelmaczonek) [inline core/kernel/persistence/smoothsql/search/ComplexSearchService.php:254]: Nitpick: We don't have any const for such strings?

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**